### PR TITLE
Default branch for Thunderbord is now 'comm-central' (fixes #247)

### LINKED ISSUE #247: Default branch for Thunderbird should be comm-central and not mozilla-central
Trying to download the latest nightly of Thunderbird fails because mozdownload assume mozilla-central:

> $ mozdownload -t daily -a thunderbird
>   INFO | Retrieving the build status file from https://ftp.mozilla.org/pub/mozilla.org/thunderbird/nightly/latest-mozilla-central/

We might need a new mapping which specified the default branch for each application separately.

### PR DESCRIPTION
**Pull request by TLemur**
Sorry for the mess everyone.
This PR fixes issue #247 and it replaces PR #285 where I messed a lot with the tree.

### CODE REVIEW

**Comment by TLemur** [comment]
@whimboo is everything ok with this?

File: mozdownload/scraper.py [hunk 1]
```diff
@@ -384,6 +388,10 @@ def __init__(self, branch='mozilla-central', build_id=None, date=None,
     def get_build_info(self):
         """Defines additional build information"""
 
+        # Find branch by application
+        if self.branch is None:
+            self.branch = DEFAULT_BRANCHES[self.application]
```
Comment (whimboo) [inline mozdownload/scraper.py]: Why can this not be done directly in `__init__()`?

File: mozdownload/scraper.py [hunk 2]
```diff
@@ -55,6 +55,10 @@
                            'win32': 'exe',
                            'win64': 'exe'}
 
+DEFAULT_BRANCHES = {'b2g': 'mozilla-central',
+                    'firefox': 'mozilla-central',
+                    'thunderbird': 'comm-central'}
```
Comment (whimboo) [inline mozdownload/scraper.py]: I don't like that we hard-code the application names twice. There is already the `APPLICATIONS` variable available. We should be able to change it to a hash with the application name as key and the appropriate branch as value.

**Comment by whimboo** [comment]
I made some comments but in general this looks good, yes! Let me know when you committed your updates.

File: mozdownload/scraper.py (hunk 1 above)
Comment (TLemur) [inline mozdownload/scraper.py]: @whimboo This is because self.application is not defined in __init__()

File: mozdownload/scraper.py (hunk 2 above)
Comment (TLemur) [inline mozdownload/scraper.py]: Agreed.

File: mozdownload/scraper.py (hunk 1 above)
Comment (whimboo) [inline mozdownload/scraper.py]: It is but in the base scraper class: https://github.com/TLemur/mozdownload/blob/issue_247/mozdownload/scraper.py#L129

Comment (TLemur) [inline mozdownload/scraper.py]: If I enter anything meaningful to `__init__()`, various tests fail.

File: mozdownload/scraper.py [hunk 3]
```diff
@@ -776,6 +782,10 @@ def __init__(self, branch='mozilla-central', build_number=None, date=None,
     def get_build_info(self):
         "Defines additional build information"
 
+        # Find branch by application
+        if self.branch is None:
+            self.branch = APPLICATIONS[self.application]
```
Comment (whimboo) [inline mozdownload/scraper.py]: Well, you are adding this code here to the `DailyScraper` class, but it really has to be part of the `Scraper` class to be available for each and every scraper. Do this e.g after line 130 when `self.application` gets set.

Comment (whimboo) [inline mozdownload/scraper.py]: Oh, please drop my comment from above. The base Scraper class doesn't need the branch property. So what you need to do here is to have this code in the `__init__` method of `DailyScaper` but after line 384 which initializes the `Scraper` class first.

**Comment by whimboo** [comment]
@TLemur do you have an update for us?

**Comment by TLemur** [comment]
Hi, I'm temporarily out of city for my vacation. I'll be back on Aug. 10.

Отправлено с моего HTC

----- Reply message -----
От: "Henrik Skupin" <notifications@github.com>
Кому: "mozilla/mozdownload" <mozdownload@noreply.github.com>
Копия: "TLemur" <tim_tim2000@mail.ru>
Тема: [mozdownload] Default branch for Thunderbord is now &apos;comm-central&apos; (fixes #247) (#286)
Дата: чт, авг 6, 2015 17:27
@TLemur do you have an update for us?

&mdash;

**Comment by whimboo** [comment]
I see. In that case enjoy your vacation!

**Comment by TLemur** [comment]
I'm back now. What do I do to continue working on this?

**Comment by whimboo** [comment]
Please check my latest comments from June 22nd about what's left to do.

**Comment by whimboo** [comment]
@TLemur looks like we forgot about this PR. Will you have the time to finish it off?

**Comment by whimboo** [comment]
Closing out PR until it can be continued.

**Pull request closed without merging**